frontend/src/components/visual-editor/v2/Diagrams.tsx

```diff
@@ -429,6 +444,56 @@ const Diagrams = () => {
       deleteDialogCtl.closeDialog();
     }
   };
+  const onMove = async (newCategoryId?: string) => {
+    if (!newCategoryId) {
+      return;
+    }
+
+    const ids = moveDialogCtl?.data;
+
+    if (ids) {
+      for (const blockId of ids) {
+        const block = getBlockFromCache(blockId);
+
+        await updateBlock(
+          {
+            id: blockId,
+            params: {
+              category: newCategoryId,
+            },
+          },
+          {
+            onSuccess() {
+              updateCachedBlock({
+                id: blockId,
+                payload: {
+                  ...block,
+                  category: newCategoryId,
+                },
+                strategy: "overwrite",
+              });
```

Comment:
Since we remove queries anyway, I don't think we need to update the cached block ?

Reply:
Correct. I removed that extra block of code. Thanks :pray: